examples/chef/create_docker.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2020 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+import optparse
+import tarfile
+import shutil
+import re
+import docker
+
+client = docker.from_env()
+
+parser = optparse.OptionParser()
+parser.add_option("-c", "--commit_sha", type='string',
+                  action='store',
+                  dest="commit_sha")
+parser.add_option("-s", "--short_sha", type='string',
+                  action='store',
+                  dest="short_sha")
+parser.add_option("-r", "--revision_id", type='string',
+                  action='store',
+                  dest="revision_id")
+parser.add_option("-b", "--build_id", type='string',
+                  action='store',
+                  dest="build_id")
+parser.add_option("-i", "--image_name", type='string',
+                  action='store',
+                  dest="image_name")
+parser.add_option("-w", "--working_dir", type='string',
+                  action='store',
+                  dest="working_dir")
+parser.add_option("-t", "--tar_path", type='string',
+                  action='store',
+                  dest="tar_path")
+parser.add_option("-p", "--platform", type='string',
+                  action='store',
+                  dest="platform")
+
+options, _ = parser.parse_args(sys.argv[1:])
+
+out_directory = f'{options.working_dir}/out'
```

Comment:
The script seems to strongly assume a structure for the working directory. Is working directory really a proper name given that it must contain a devices folder? should devices folder and output folder maybe be separate command line options instead?

Reply:
This script relies on the structure that is in that examples/chef directory. Similar to the chef.py script, I decided to remove this from the arguments and store it in a global variable. Let me know if this works.